From 6aa1d98fa53773cecafd54b925703ae8a9f1c63b Mon Sep 17 00:00:00 2001
From: Cheng Zhao <zcbenz@gmail.com>
Date: Fri, 1 Mar 2024 21:20:53 +0900
Subject: [PATCH] tools: fix installing node with shared mode

Fix a bug caused by https://github.com/nodejs/node/pull/50737 that,
make install fails for --shared mode.

PR-URL: https://github.com/nodejs/node/pull/51910
Reviewed-By: Richard Lau <rlau@redhat.com>
Reviewed-By: Luigi Pinca <luigipinca@gmail.com>
---
 tools/install.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/install.py b/tools/install.py
index 196bfdf86045b4..8f816dc1c1ed50 100755
--- a/tools/install.py
+++ b/tools/install.py
@@ -176,7 +176,7 @@ def files(options, action):
       try_symlink(options, so_name, link_path)
     else:
       output_lib = 'libnode.' + options.variables.get('shlib_suffix')
-      action(options, [os.path.join(output_prefix, output_lib)],
+      action(options, [os.path.join(options.build_dir, output_lib)],
              os.path.join(options.variables.get('libdir'), output_lib))
 
   action(options, [os.path.join(options.v8_dir, 'tools/gdbinit')], 'share/doc/node/')
